src/RestQuery.js

```diff
@@ -501,6 +510,29 @@ RestQuery.prototype.replaceEquality = function() {
   }
 }
 
+// Runs any beforeFind triggers against this operation.
+RestQuery.prototype.runBeforeTrigger = function() {
+  if (this.response) {
+    return;
+  }
+
+  // Avoid doing any setup for triggers if there is no 'beforeFind' trigger for this class.
+  if (!triggers.triggerExists(this.className, triggers.Types.beforeFind, this.config.applicationId)) {
+    return Promise.resolve();
+  }
+
+  return triggers.maybeRunQueryTrigger(triggers.Types.beforeFind, this.className, this.restWhere, this.restOptions,
+    this.config, this.auth, this.isGet).then((result) => {
```

Comment:
not a big fan of mutating the state here... this may lead to unpredictable behaviours in testing etc... I'd rather keep in the rest.js.

Reply:
also duplicates the call to handleOptions...

I don't believe it should be refactored this way.

Reply:
The same premise is used in ParseWrite.js where a runBeforeTrigger function mutates the state if the object is changed.

However, you make an interesting point that handleOptions function is called twice. Maybe make the handleOptions function apart of the execute function? That way it will only be called once after the runBeforeTrigger?

Regardless, I think we should keep RestQuery and RestWrite consistent.

Reply:
Take a look [here](https://github.com/parse-community/parse-server/blob/master/src/RestWrite.js#L144) for what I'm referring to in the RestWrite class. It has the same function as I'm implementing for RestQuery. 

This is what I mean when we should keep these two classes consistent.

Reply:
The mutating of the state for hooks in RestWrite is primarily to modify the object being saved/updated. I don't believe we ever intended to modify the queries dynamically in any of the finds. Although I do like the idea of keeping the two layouts standardized by moving this in, I don't believe we should stretch beyond that and start introducing query modification behavior. If we want to consider that I would like to do so in a separate PR, keeping this one as strictly a structural adjustment 👍 .

Reply:
Ehh, although looking back it what you have added here is about the same as to what we had before...hmm...@flovilmart I can't think off the top of my head, but modifying the state was present before in the same form that it has been presented here. Was there a reason for this or was it something vestigial?